chembl_db/generate_chembl_db.py

```diff
@@ -0,0 +1,498 @@
+    ap.add_argument("--out-csv", default="chembl_db.csv", help="Final CSV output (default: chembl_db.csv)")
+    ap.add_argument("--skip-ccd", action="store_true", help="Skip running the CCD script (assumes --ccd-csv exists)")
+    ap.add_argument("--drop-missing-pdb", action="store_true", help="Drop rows with NaN in pdb_id after CCD merge")
+    ap.add_argument("--pfam-workers", type=int, default=24, help="Threads for UniProt Pfam step (default: 24)")
+    ap.add_argument("--pfam-timeout", type=int, default=45, help="Per-request timeout seconds (default: 45)")
+    ap.add_argument("--pfam-retries", type=int, default=3, help="Max retries for UniProt calls (default: 3)")
+    ap.add_argument("--log-dir", default="logs", help="Directory to store logs (default: logs)")
+    ap.add_argument("--log-level", default="INFO", help="Logging level (DEBUG, INFO, WARNING, ERROR)")
+    ap.add_argument("--no-console", action="store_true", help="Do not echo logs to console (file only)")
+    return ap.parse_args()
+
+
+def main() -> int:
+    """
+    Run the complete pipeline according to CLI arguments with structured logging.
+
+    Returns
+    -------
+    int
+        Zero on success; non-zero if any step fails.
+    """
+    args = parse_args()
+
+    logger = setup_logger(Path(args.log_dir), level=args.log_level, console=not args.no_console)
+
+    chembl_db = Path(args.chembl_sqlite)
+    ccd_script = Path(args.ccd_script)
+    workdir = Path(args.workdir)
+    ccd_csv = Path(args.ccd_csv)
+    out_csv = Path(args.out_csv)
+
+    # 1) Query ChEMBL SQLite
+    results = load_chembl_results(chembl_db, logger)
+
+    # 2) Run CCD script (unless skipped)
+    if not args.skip_ccd:
+        rc = run_ccd_script(sys.executable, ccd_script, workdir=workdir, ccd_csv=ccd_csv, logger=logger)
+        if rc != 0:
+            logger.error("CCD script failed with return code %s", rc)
+            return rc
+    else:
+        logger.info("[2/5] Skipping CCD run (using existing CSV)")
+
+    # 3) Read CCD CSV and merge pdb_id by InChIKey
+    ccd_map = read_ccd_map(ccd_csv, logger)
+    results = add_pdb_id_by_inchikey(results, ccd_map, logger)
+    if args.drop_missing_pdb:
+        before = len(results)
+        results = results.dropna(subset=["pdb_id"]).reset_index(drop=True)
+        logger.info("    dropped rows without pdb_id: %s", f"{before - len(results):,}")
+
+    # 4) Resolve Pfam IDs per protein and merge
+    results = add_pfam_ids_by_uniprot(
+        results,
+        max_workers=args.pfam_workers,
+        timeout=args.pfam_timeout,
+        retries=args.pfam_retries,
+        logger=logger,
+    )
+
+    # 5) Write final CSV
+    out_csv.parent.mkdir(parents=True, exist_ok=True)
+    results.to_csv(out_csv, index=False)
+    logger.info("[5/5] Wrote final CSV: %s", out_csv)
+
+    logger.info("Pipeline finished successfully.")
+    return 0
+
+
+if __name__ == "__main__":
+    raise SystemExit(main())
```

Comment:
```suggestion
    main()
```

Don't forget that you have to change the function name

Reply:
Ok